File: go/3_LongestSubstringWithoutRepeatingCharacters_test.go
package main

import (
    "github.com/stretchr/testify/assert"
    "testing"
)

func TestLengthOfLongestSubstring(t *testing.T) {
    assert.Equal(t,
        3,
        lengthOfLongestSubstring("abcabcbb"),
    )
    assert.Equal(t,
        1,
        lengthOfLongestSubstring("bbbbb"),
    )
    assert.Equal(t,
        3,
        lengthOfLongestSubstring("pwwkew"),
    )
    assert.Equal(t,
        3,
        lengthOfLongestSubstring("aabaab!bb"),
    )
    assert.Equal(t,
        1,
        lengthOfLongestSubstring(" "),
    )
    assert.Equal(t,
        2,
        lengthOfLongestSubstring("au"),
    )
}

File: python/3_LongestSubstringWithoutRepeatingCharacters_test.py
import unittest


class TestSolution(unittest.TestCase):

    def test_LengthOfLongestSubstring(self):
        Solution = __import__('3_LongestSubstringWithoutRepeatingCharacters').Solution
        self.assertEqual(
            3,
            Solution().lengthOfLongestSubstring("abcabcbb")
        )
        self.assertEqual(
            1,
            Solution().lengthOfLongestSubstring("bbbbb")
        )
        self.assertEqual(
            3,
            Solution().lengthOfLongestSubstring("pwwkew")
        )
        self.assertEqual(
            3,
            Solution().lengthOfLongestSubstring("aabaab!bb")
        )
        self.assertEqual(
            1,
            Solution().lengthOfLongestSubstring(" ")
        )
        self.assertEqual(
            2,
            Solution().lengthOfLongestSubstring("au")
        )
